readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/hive/HiveClient.java

```diff
@@ -1,79 +1,51 @@
 package com.criteo.hadoop.garmadon.hdfs.hive;
 
-import org.apache.commons.lang.exception.ExceptionUtils;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.commons.lang3.tuple.Pair;
 import org.apache.parquet.schema.MessageType;
 import org.apache.parquet.schema.Type;
-import org.apache.thrift.transport.TTransportException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.sql.Connection;
-import java.sql.DriverManager;
 import java.sql.SQLException;
-import java.sql.Statement;
 import java.util.stream.Collectors;
 
 public class HiveClient {
     private static final Logger LOGGER = LoggerFactory.getLogger(HiveClient.class);
 
-    private final String database;
-    private final String jdbcUrl;
-    private Connection connection;
-    private Statement stmt;
+    private final HiveQueryExecutor executor;
+    private final Set<Pair<String, String>> createdPartitions = new HashSet<>();
```

Comment:
even if it takes a lot of time to fill memory with these, we should find a way to clean old entries

Reply:
I was unsure whether this was premature optimization, but since it seems fairly simple to implement, I did.